packages/abi-coder/src/types/IGetCoder.ts

```diff
@@ -0,0 +1,19 @@
+import type { ResolvedAbiType } from '../ResolvedAbiType';
+import type { Coder } from '../encoding/coders/AbstractCoder';
+
+import type { TEncodingOptions } from './TEncodingOptions';
+
+/**
+ * A function that can be used to obtain spec adhering coders.
+ *
+ * @param resolvedAbiType - the resolved type to return a coder for.
+ * @param options - options to be utilized during the encoding process.
+ */
+export type TGetCoderFn = (resolvedAbiType: ResolvedAbiType, options?: TEncodingOptions) => Coder;
```

Comment:
Same as [here](https://github.com/FuelLabs/fuels-ts/pull/1780/files#r1497763483), no need for the `T` prefix.